src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/NumberHandlingTests.cs

```diff
@@ -88,6 +89,11 @@ private static void RunAsRootTypeTest<T>(List<T> numbers)
 
         private static string GetNumberAsString<T>(T number)
         {
+            if (OperatingSystem.IsAndroid() && RuntimeInformation.ProcessArchitecture == Architecture.X86)
+            {
+               return number.ToString();
+            }
+            
```

Comment:
is this problem specific to JSON? Where is this difference coming from?

Reply:
Issue is related to float to string conversion.